### PR TITLE
Esp participation

### PR DESCRIPTION
**Pull request by joeeames**
## Description

This PR adds links and adjustments for participation in the OpenJS Ecosystem Sustainability Program (ESP). This includes a new top nav link, a warning on the download page when downloading EOL versions, and information on the "previous releases" page including a link to HeroDevs. 

Once merged, the graphic on the Previous Releases page will be updated, and the core node.js README on the github repo will be updated to summarize some of this information. This will be done with separate PR's.

## Validation

1. New top nav pointing to the Previous Releases page 
2. a warning box on the download page when selecting a version before 18
3. new text on the Previous Releases page

### Check List

- [ X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [ X] I have run `npm run format` to ensure the code follows the style guide.
- [ X] I have run `npm run test` to check if all tests are passing.
- [ X] I have run `npx turbo build` to check if the website builds without errors.
- [ X] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (mikeesto) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: Missing styling here for dark mode - it's currently unreadable

File: apps/site/navigation.json
Comment (mikeesto) [inline apps/site/navigation.json]: This needs to be a relative link. It's also unclear to me why we need to add previous releases to the top nav bar?

File: packages/i18n/locales/fr.json
Comment (mikeesto) [inline packages/i18n/locales/fr.json:1]: We use CrowdIn for translations, the translations in this file & others can be removed

**Review by mikeesto — CHANGES_REQUESTED** [review]
It's unclear to me how this fits in with https://github.com/nodejs/nodejs.org/issues/7037 ? If the underlying motivation is to market support for EOL versions through HeroDevs, that can be a short spiel on the Node.js Releases page. The changes to the Download page seem unproductive given it's about to be redesigned

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx:36]: Maybe we should put these styles in a css module to make them cleaner?

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: on project we use `semver` that export a function call `semverParse`

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: You use `import Link from '@/components/Link';` to automatically use right local (so no need to use the `useLocal` hook)

**Review by AugustinMauroy — COMMENTED** [review]
+1 for this mention; +1 to display it as "alert box"

**Comment by ovflowd** [comment]
> [ X] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md?rgh-link-date=2024-10-03T22%3A51%3A24Z) and made commit messages that follow the guideline.

Have you actually read the guidelines? 🤔 -- because some of the changes you've done here, such as translations are mentioned there 🤔

**Review by ovflowd — COMMENTED** [review]
Overall I'm fine with the change, although not ideal, but you definitely need to use the existing components from our design system please.

**Comment by AugustinMauroy** [comment]
>  but you definitely need to use the existing components from our design system please.

I had the same feeling. But we don't have any red flash components at the moment, with the exception of `Banner`. If not there's `Stability` but that's an API component that hasn't been implemented yet.

**Comment by ovflowd** [comment]
> > but you definitely need to use the existing components from our design system please.
> 
> I had the same feeling. But we don't have any red flash components at the moment, with the exception of `Banner`. If not there's `Stability` but that's an API component that hasn't been implemented yet.

Banner component is fine.

File: apps/site/navigation.json
Comment (joeeames) [reply apps/site/navigation.json]: thanks for calling this out. I did want to use a relative link, but must have just looked at the closest neighbor. will address.

As to having the link in the top nav bar, this is part of the requirements for participation in the ESP program. The page showing the support and EOL policy for the project needs to be easy to find for full participation, generally meaning part of the main navigation.

**Comment by joeeames** [comment]
thanks for all the review. I apologize for what I missed. Although I did read the contributing guidelines @ovflowd I did misunderstand the translation piece. I'll get changes made here and resubmitted.

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (joeeames) [reply apps/site/components/Downloads/Release/ReleaseCodeBox.tsx:36]: @AugustinMauroy Augustin, not sure what is meant by "alert box" can you clarify?

Also, not sure if changing to a CSS module is required or not. I'm not familiar with what that would entail.

Comment (AugustinMauroy) [reply apps/site/components/Downloads/Release/ReleaseCodeBox.tsx:36]: my question about css module is more for the website team. And alert box mean red box. But you should use Banner component like Claudio said in this [comment](https://github.com/nodejs/nodejs.org/pull/7086#issuecomment-2395559841)

**Comment by AugustinMauroy** [comment]
I'm -1 to have a new entry on nav bar because it's not 100% usefull. It also introduces a strange effect on the navbar because it hasn't been designed to support two entries that have the same ‘section’.

Capture d’écran 2024-10-15 à 09 26 12

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]:
```suggestion
```

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]:
```suggestion
```

**Review by AugustinMauroy — COMMENTED** [review]
Some thing can be clean

**Comment by joeeames** [comment]
changes have been made to address all requested items (I believe)

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (joeeames) [reply apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: fixed

**Comment by joeeames** [comment]
> I'm -1 to have a new entry on nav bar because it's not 100% usefull. It also introduces a strange effect on the navbar because it hasn't been designed to support two entries that have the same ‘section’.
> 
> Capture d’écran 2024-10-15 à 09 26 12

We need to find a resolution here. Participation in the ESP program is about making sure that users who need to understand what versions are supported and what are not, and consequently find out about solutions for versions that are not still supported, need a simple and easy way to find the page documenting this information. Currently it's buried under "about" and then "previous releases" which isn't at all straightforward for someone wanting to find out about support. I get the comment about it being "100% useful" but for someone trying to find out if the version they're on is supported, these links are critical. 

Some other options are to create an entirely new page as we've done with many other projects, but the previous releases page is by far more suited for this. Or to remove it from the about section and put it in its own section? We're open to anything that fits the criteria of "easy to find". So any ideas are welcome here.

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (AugustinMauroy) [inline apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: where does ‘18.0.0’ come from? and why ?

**Comment by AugustinMauroy** [comment]
I don't agree with you that it's buried in the about section. It's fine like that. Because the section above is a navbar whose purpose is to navigate between sections and NOT to go to a specific article.
I've looked elsewhere and no-one puts ‘version support’ in the nav. I've checked for Deno, .net, java, php, ruby.

File: apps/site/components/Downloads/Release/ReleaseCodeBox.tsx
Comment (joeeames) [reply apps/site/components/Downloads/Release/ReleaseCodeBox.tsx]: yeah, sadly it's hardcoded. i didn't see that this was encoded anywhere in metadata, but 18 is the oldest "supported" version as of today. This is definitely not ideal as it would need to be updated whenever new versions fall out of support.

**Comment by joeeames** [comment]
> I don't agree with you that it's buried in the about section. It's fine like that. Because the section above is a navbar whose purpose is to navigate between sections and NOT to go to a specific article. I've looked elsewhere and no-one puts ‘version support’ in the nav. I've checked for Deno, .net, java, php, ruby.

Sorry I wasn't meaning "projects in general" but "projects we've worked with". Meaning the purpose of the ESP is to improve the use case for users for whom support matters. Part of that is making the support information easy to find. So to clarify, on the other projects that HeroDevs has worked with (angular, jquery, bootstrap, vue 2, ESLint, etc) we've added links to this information usually in the main nav or on the home page. This is obviously not the only way to do this, but we work to make this page easy to find. 

It's very subjective of course, I just try to put myself in the shoes of someone who may not know a site intimately, can I find the information I'm looking for and if not what change would make it easy to find. Since I personally do this as I look to see what changes to recommend to make, I think I'm a good proxy for this user profile. So when I went looking for the information on the Node website I definitely had some trouble finding it. Again, this is very subjective. 

Thought I'd throw in some screenshots of ways some other project have addressed this. Two of these three did NOT go with a new top nav item but instead added something to the home page.

ESLINT:
image

jQuery:
image

Bootstrap:
image

**Comment by marco-ippolito** [comment]
@mikeesto can you please take a look again? Please let us know how to move forward

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by bmuenzenmeyer** [comment]
I'm unsure why https://github.com/nodejs/nodejs.org/issues/7037 was first referenced, the OP doesn't seem to.

To be clear - the project and website maintainers fully support this effort. I'm happy to help land it to satisfaction of all parties.
I briefly looked into the rendering concerns that Augustin (rightly) mentioned within [this comment](https://github.com/nodejs/nodejs.org/pull/7086#issuecomment-2413107188) - I think that's the only real blocker from a UX perspective.

I'll try to move things along by end of weekend.

**Review by mikeesto — APPROVED** [review]
Thanks for taking our feedback on board @joeeames . Perhaps the nav bar item could be named "Releases" instead of "Version Support" ? I agree with others that it would be good to fix the UX here too before this lands

File: apps/site/components/Common/ActiveLink/index.tsx
Comment (bmuenzenmeyer) [inline apps/site/components/Common/ActiveLink/index.tsx:32]: if this doesn't merge by morning, i may add unit tests for the entire `allowSubpath` part of the ternary, which has no test coverage. it requires mocking `usePathname`

File: apps/site/next.constants.mjs
Comment (bmuenzenmeyer) [inline apps/site/next.constants.mjs:180]: moving this to a ENV VAR or constant makes the magic string more obvious. with even versioning being the important milestone for prod usage, this is only really in need of touch twice a year, right?

we could automate it with this, as other portions of the code do

```
import getReleaseData from '@/next-data/releaseData';
```

**Review by bmuenzenmeyer — APPROVED** [review]
push a commit onto the branch, hope that's okay with you @joeeames

**Comment by bmuenzenmeyer** [comment]
failing test is resolved on `main` via https://github.com/nodejs/nodejs.org/commit/851b085af21fe74612eccae9094803574c50e39c

File: apps/site/components/Common/ActiveLink/index.tsx
Comment (bmuenzenmeyer) [reply apps/site/components/Common/ActiveLink/index.tsx:32]: added via https://github.com/nodejs/nodejs.org/pull/7086/commits/9847aef066a21499174b84b59e8876945601195a

**Pull request merged**
